### PR TITLE
[AIRFLOW-6699] Parameterize weekday_sensor tests

### PR DESCRIPTION
**Pull request by coopergillan**
Using `pytest`, consolidate the "True" or "happy path" tests into one parameterized test that uses a new TRUE_TESTS_CONFIG to drive test names and the `week_day` parameter.

Update all tests to use `pytest`, including regex matching on the failing tests as needed.

---
Issue link: [AIRFLOW-6699](https://issues.apache.org/jira/browse/AIRFLOW-6699)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: tests/sensors/test_weekday_sensor.py [hunk 1]
```diff
@@ -33,130 +33,91 @@
...
+    session.query(TaskFail).filter_by(
+        dag_id=TEST_DAG_ID).delete()
+    session.commit()
+    session.close()
+
+
+@pytest.mark.parametrize("weekday", TRUE_TESTS_CONFIG.values(), ids=list(TRUE_TESTS_CONFIG.keys()))
+def test_weekday_sensor_true(dag, weekday, tear_down):
+    op = DayOfWeekSensor(
+        task_id="weekday_sensor_check_true",
```
Comment (mik-laj) [inline tests/sensors/test_weekday_sensor.py:68]: It is necessary to use classes in tests. More information: https://lists.apache.org/thread.html/1e4df7d4b0cd9b2d2bb76a3336471aa85e852545dd41ada6d4e461b8%40%3Cdev.airflow.apache.org%3E (question 5)

Comment (coopergillan) [reply tests/sensors/test_weekday_sensor.py:68]: Ah, that makes sense. Thank you for the heads up 👍

**Comment by coopergillan** [comment]
Closed in favor of https://github.com/apache/airflow/pull/7316.

File: tests/sensors/test_weekday_sensor.py (hunk 1 above)
Comment (coopergillan) [reply tests/sensors/test_weekday_sensor.py:68]: This helped me learn about how tests **are** parameterized here in `airflow`. Went with a different approach over in #7316.

**Pull request closed without merging**